[java] Fix Issue 1343: Update CommentDefaultAccessModifierRule

CrazyUnderdog opened the pull request:
Before submitting a PR, please check that:
 - [x] The PR is submitted against `master`. The PMD team will merge back to support branches as needed.
 - [x] `./mvnw clean verify` passes. This will [build](https://github.com/pmd/pmd/blob/master/BUILDING.md) and test PMD, execute PMD and checkstyle rules. [Check this for more info](https://github.com/pmd/pmd/blob/master/CONTRIBUTING.md#code-style)

**PR Description:**

Fixes #1343, the class CommentDefaultAccessModifierRule now extends AbstractIgnoredAnnotationRule. Also added a small comment about the ignored nodes in CommentDefaultAccessModifierRule.

pmd-test commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This changeset introduces 0 ne...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">This changeset introduces 0 new violations and 0 new errors,<br />
removes 0 violations and 2 errors. <a href="http://chunk.io/pmd/dfe23203e0ed4cf296151ca8f51a0ce9/diff/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

CrazyUnderdog commented:
Just a quick question, that I understand this correctly:

By using the overridden *defaultSuppressionAnnotations* method, I can declare the String "VisibleForTesting" to be ignored by default.

Does that mean that all I have to do in *hasNoVisibleForTestingAnnotation

jsotuyod commented on pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/CommentDefaultAccessModifierRule.java:
```diff
@@ -41,6 +41,14 @@
+    @Override
+    protected Collection<String> defaultSuppressionAnnotations() {
+        Collection<String> ignoredStrings = new ArrayList<>();
+        ignoredStrings.add("VisibleForTesting");
```
you should use fully qualified class names where for proper type matching.

the ones we probably care about are:
`com.google.common.annotations.VisibleForTesting`
`android.support.annotation.VisibleForTesting`

jsotuyod commented on pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/CommentDefaultAccessModifierRule.java:
```diff
@@ -41,6 +41,14 @@
 
     public CommentDefaultAccessModifierRule() {
         definePropertyDescriptor(REGEX_DESCRIPTOR);
+        defaultSuppressionAnnotations();
```
there is no need to call this method here

adangel replied:
Yes, this is called already by the super class, when it is defining the property:

https://github.com/pmd/pmd/blob/7edaf78e4a61a7e49d8ca67e21fbed4e01fb9ffd/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/AbstractIgnoredAnnotationRule.java#L18

CrazyUnderdog commented:
I can take a look at this at the weekend. However I want you to know that I did this pull request as a part of an event, and can't put much time into this anymore.

CrazyUnderdog commented:
Okay, for whatever reason, my IDE does not want to import your codestyle, so that check still fails.

I've removed the unnecessary method all and added the full class paths to the ignoredStrings collection.

adangel commented:
@CrazyUnderdog Thanks for your time! I'll take it from here

adangel commented:
@CrazyUnderdog Thanks for the PR, merged! :tada:
